packages/shared/lib/core/wallet/actions/claimActivity.ts

```diff
@@ -13,26 +12,15 @@ export async function claimActivity(activityId: string, data: ITransactionActivi
     const _activeProfile = get(activeProfile)
     try {
         updateActivityDataByActivityId(account.id, activityId, { type: ActivityType.Transaction, isClaiming: true })
-        const results = await account.claimOutputs([data.outputId])
-        if (results.length > 0) {
-            const transactionId = results[0].transactionId
-            updateActivityDataByActivityId(account.id, activityId, {
-                type: ActivityType.Transaction,
-                claimingTransactionId: transactionId,
-            })
-        } else if (_activeProfile.type === ProfileType.Ledger) {
-            // TODO: remove these clauses after wallet.rs implemented errors
-
-            // Wallet.rs doesn't throw an error for claiming (multiple) outputs.
-            // Multiple outputs could be claimed in multiple transactions and
-            // it's possible for some to fail and others to succeed.
-            throw new Error(LedgerError.DeniedByUser)
-        } else {
-            throw new Error(localize('error.send.cannotClaimTwice'))
-        }
+        const result = await account.claimOutputs([data.outputId])
+        const transactionId = result.transactionId
+        updateActivityDataByActivityId(account.id, activityId, {
+            type: ActivityType.Transaction,
+            claimingTransactionId: transactionId,
+        })
     } catch (err) {
-        if (err.message === LedgerError.DeniedByUser) {
-            handleLedgerError(err.message)
+        if (_activeProfile.type === ProfileType.Ledger) {
+            handleLedgerError(err.error)
```

Comment:
i think we need another check here, because not all errors (if on a ledger acc) are ledger errors, most of them will still be wallet.rs error and therefore need to be handled with `handleError`